blog/_posts/2017-09-12-Issue-253.md

```diff
@@ -0,0 +1,134 @@
+3. We know what is server, browser, database, models, migrations, HTTP requests, querysets, filters and all these fancy words around we did not know how they are connected before.
+4. And finally: This Issue is not as small as it looks like!
+
+If you want to read more about this, go to the Foodsaving blog and read the [Fairy Tale Version](https://blog.foodsaving.world/2017/09/01/railsgirls-fairy-tale.html) of that story.
+
+<a href="https://blog.foodsaving.world/2017/09/01/railsgirls-fairy-tale.html"> <img src="/img/blog/2017/Issue253.jpg" width="300"></a><br>
+_Graphic by Ines D. Guett_
+
+
+#### We Have Built (Status: 6. September 2017):   
+- a new API endpoint Feedback with HTTP methods GET, POST  
+- integration API tests for Feedback that ensure that the user gets the correct GET/POST output  
+- validation for field ```about``` and set the field ```given_by``` as read-only
+- unit model tests for model Feedback
+
+<br>
+
+Our hardest part on our Issue was to fix this test:
+```python
+    def test_list_feedback_works_as_collector(self):    
+        """
+        Collector is allowed to see list of feedback.
+        """
+        self.client.force_login(user=self.collector)
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+        self.assertEqual(len(response.data), 2)
+```
+This test should return two instances of our test object and check that there are exactly 2 instances in the test database. There is also a difference between a test database and live database. We discussed this many times until we could fully understand :)
+
+<br>
+
+<script id="infogram_0_a09df277-2245-4bca-bec9-922b039710e6" title="Welcome: Your first project" src="https://e.infogram.com/js/dist/embed.js?yjt" type="text/javascript"></script><div style="padding:8px 0;font-family:Arial!important;font-size:13px!important;line-height:15px!important;text-align:center;border-top:1px solid #dadada;margin:0 30px"><a href="https://infogram.com/a09df277-2245-4bca-bec9-922b039710e6" style="color:#989898!important;text-decoration:none!important;" target="_blank">Welcome: Your first project</a><br><a href="https://infogram.com" style="color:#989898!important;text-decoration:none!important;" target="_blank" rel="nofollow">Infogram</a></div>
+
+<br>
+
+#### We Still Want to Do:
+- add HTTP method PATCH to Feedback
+- change field ```weight``` to FloatField
+- add filters to our Feedback + some other limitations
+- refacture
+- writing a Foodsaving-backend beginner guide. We already started [here](https://github.com/mddemarie/Foodsaving-Documentation/blob/master/foodsaving-backend-code-guide.md).
+
+<br>
+
+## Hackathon & Community
+
+We attended HackDays in ImmobillienScount 11.07.-13.07. and we have got very motivated with Stefan: "We can make it!"
```

Comment:
We have attended HackDays in ImmobillienScount 11.07.-13.07. and we have got very motivated with Stefan: "We can make it!"
 +
 +Foodsaving founders invited us to their **Hackathon: 14.07.-16.07**. We've met incredibly smart people here - Tilmann Becker  (@tiltec), Nick Sellen (@nicksellen) and Janina Abels (@djahnie) who put their time & energy into making the saving a food waste possible.